torch_xla/experimental/xla_sharding.py

```diff
@@ -64,12 +65,20 @@ def size(self):
     return np.prod(self.mesh_shape)
 
   def shape(self):
+    if self.axis_names is None:
+      return OrderedDict(
+          (dim, size) for dim, size in enumerate(self.mesh_shape))
     return OrderedDict(
         (name, size) for name, size in zip(self.axis_names, self.mesh_shape))
 
   def get_logical_mesh(self):
     return self.device_ids.reshape(self.mesh_shape)
 
+  def get_axis_name_idx(self, name: str) -> int:
+    if name not in self.axis_names:
+      return None
+    return [i for i, _ in enumerate(self.axis_names) if _ == name].pop()
```

Comment:
Can we `return self.axis_names.index(name)`?

Reply:
Nice, thanks!